tests/lib/prepare.sh

```diff
@@ -450,10 +469,15 @@ prepare_classic() {
 
         snap list | grep core
 
-        systemctl stop snapd.{service,socket}
-        # repack and also make a side copy of the core snap
-        update_core_snap_for_classic_reexec "$TESTSTMP/core_snap"
-        systemctl start snapd.{service,socket}
+        # No need to update the core snap when the snap pkg is not being built from local changes
```

Comment:
I think the comment should be reworded to better describe the motivation here. With reexec, and on classic, the snapd snap is preferred over the core snap for reexecution target, so to be as close as possible to the actual real life scenarios, we only update the snapd snap. The tests alreday ensure that snapd snap is installed.

Reply:
updated